cpp/bench/spatial/selection.cu

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2019-2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <common/benchmark.hpp>
+#include <raft/spatial/knn/knn.cuh>
+
+#include <raft/random/rng.hpp>
+#include <raft/sparse/detail/utils.h>
+
+#include <rmm/mr/device/per_device_resource.hpp>
+#include <rmm/mr/device/pool_memory_resource.hpp>
+
+namespace raft::bench::spatial {
+
+struct params {
+  int n_inputs;
+  int input_len;
+  int k;
+  int select_min;
+};
+
+template <typename KeyT, typename IdxT, raft::spatial::knn::SelectKAlgo Algo>
```

Comment:
Since this is a new file, we should use consistent casing/naming of template arguments. I vote for `idx_t` and `key_t` to match other functions in RAFT's nearest neighbors API.

Reply:
I don't mind your variant, but I changed this cause I thought we're trying now to stick to the style declared in [cpp/.clang-tidy](https://github.com/rapidsai/raft/blob/15eb20a4e90a2c9267dea7046dc9cb9cba3e678e/cpp/.clang-tidy#L211-L212). So I just follow the suggestions of clang-tidy to rename things.

Would you like me to change the template parameter naming here or the rules in .clang-tidy?

Reply:
Closing this with the reference to the new issue https://github.com/rapidsai/raft/issues/591 .